src/browser/ui/ReactDOMComponent.js

```diff
@@ -234,8 +247,13 @@ ReactDOMComponent.Mixin = {
           }
           propValue = CSSPropertyOperations.createMarkupForStyles(propValue);
         }
-        var markup =
-          DOMPropertyOperations.createMarkupForProperty(propKey, propValue);
+        var markup = null;
+        if (this._tag != null && this._tag.indexOf('-') >= 0) {
+          validateDangerousAttributeName(propKey);
+          markup = DOMPropertyOperations.createMarkupForCustomAttribute(propKey, propValue);
```

Comment:
I think I'd feel more comfortable if the validateDangerousAttributeName logic lived inside DOMPropertyOperations -- that way the code generating the HTML strings is the one to do the validation.